File: Assignment 9 2/app/src/main/java/com/example/tdd/MainActivity.java
package com.example.tdd;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private EditText numberEditText;
    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        numberEditText = findViewById(R.id.numberEditText);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int number = Integer.parseInt(numberEditText.getText().toString());
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                intent.putExtra("number", number);
                startActivity(intent);
            }
        });
    }
}

File: RecipeBook/app/src/main/java/com/example/recipebook/MainActivity.kt
package com.example.recipebook

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ViewGroup
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

private const val VIEW_TYPE_GROUP = 0
private const val VIEW_TYPE_FOOD = 1

class MainActivity : AppCompatActivity() {

    private val foodAdapter by lazy {
        FoodAdapter(
            layoutInflater,
            object : FoodAdapter.OnClickListener {
//                override fun onClick(foodData: FoodObject) = FoodViewHolder.OnClickListener.onItemClick(foodData)
                override fun onItemClick(foodData: FoodObject) {
                    showSelectionDialog(foodData)
                }
            }
        )
    }

    private val recyclerView: RecyclerView by lazy{
        findViewById(R.id.recycler_view) }


    /*
    override fun getItemViewType(position: Int) = when (FoodList[position]) {
        is FoodList.Group -> VIEW_TYPE_GROUP
        is FoodList.Food -> VIEW_TYPE_FOOD
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =   when (viewType) {
        VIEW_TYPE_GROUP -> {
            val view = layoutInflater.inflate(R.layout.item_food,
                parent, false)
            GroupViewHolder(view)
        }
        VIEW_TYPE_FOOD -> {
            val view = layoutInflater.inflate(R.layout.item_food, parent, false)
            FoodViewHolder(view, object :
                FoodViewHolder.OnClickListener {
                override fun onClick(foodData: FoodObject) = onClickListener.onItemClick(foodData)
            })
        }
        else -> throw IllegalArgumentException("Unknown view type requested:       $viewType")
    }
    */

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView.adapter = foodAdapter
        recyclerView.layoutManager =
            LinearLayoutManager(this, LinearLayoutManager.VERTICAL,
                false)
        foodAdapter.setData(
            listOf(
                FoodObject(
                    "Apple pie",
                    "bake the recipe for 40 minutes!",
                    FlavorType.sweet
                ),
                FoodObject(
                    "Second Test Statement",
                    "This is another description!",
                    FlavorType.savory
                )
            )
        )
    }

    private fun showSelectionDialog(foodData: FoodObject) {
        AlertDialog.Builder(this)
            .setTitle("Recipe")
            .setMessage("${foodData.description}")
            .setPositiveButton("OK") { _, _ -> }
            .show()
    }
}